ostelco-ios-client/src/extensions/UIViewController+UIActivityIndicator.swift

```diff
@@ -8,32 +8,27 @@
 
 import UIKit
 
-// TODO: Could be a good idea to not have a global variable. Rather return the view from showSpinner, then use that as a parameter to removeSpinner
 // ref: http://brainwashinc.com/2017/07/21/loading-activity-indicator-ios-swift/
-var vSpinner : UIView?
 
 extension UIViewController {
-    func showSpinner(onView : UIView) -> UIView? {
+    func showSpinner(onView : UIView) -> UIView {
         let spinnerView = UIView.init(frame: onView.bounds)
         spinnerView.backgroundColor = UIColor.init(red: 0.5, green: 0.5, blue: 0.5, alpha: 0.5)
         let ai = UIActivityIndicatorView.init(style: .whiteLarge)
         ai.startAnimating()
         ai.center = spinnerView.center
-        
+
         DispatchQueue.main.async {
             spinnerView.addSubview(ai)
             onView.addSubview(spinnerView)
         }
-        
-        vSpinner = spinnerView
-        
-        return vSpinner
+        return spinnerView
     }
-    
-    func removeSpinner() {
+
+    func removeSpinner(_ spinnerView: UIView?) {
```

Comment:
I think it would make sense to make the spinnerView parameter mandatory. Since the global spinner variable is gone, you would only call the removeSpinner function if you actually want to remove a spinner.